aten/src/ATen/core/jit_type.h

```diff
@@ -661,6 +663,15 @@ struct CAFFE2_API TensorType : public Type {
     return scalar_type_ && device_ && sizes_.isComplete() && strides_.isComplete();
   }
 
+  bool isInferredType() const {
+    return is_inferred_;
+  }
+
+  static TensorTypePtr getInferred() {
+    static auto valueInferred = TensorType::create(true);
```

Comment:
What do you think about adding a new argument to the existing `create` methods, and then changing this line to
```suggestion
    static auto valueInferred = TensorType::create(
      /*scalar_type=*/{}, /*device=*/{}, /*sizes=*/VaryingShape<ShapeSymbol>{}, /*stride=*/VaryingShape<Stride>{}, /*requires_grad=*/{}, /*undefined=*/false, /*is_inferred=*/true);
```
Then, you don't need an extra `TensorType::create(bool is_inferred)` function that does the same thing but sets `is_inferred` after creating the object.